swagger_server/controllers_local/controller.py

```diff
@@ -0,0 +1,524 @@
+from swagger_server.configuration import ons_env
+from swagger_server.controllers_local import validate
+from swagger_server.controllers_local.util import filter_falsey_values, model_to_dict
+from swagger_server.models_local.model import Business, Party, Respondent, BusinessRespondent, Address
+
+db = ons_env
+
+
+#
+# /businesses
+#
+def businesses_get(searchString=None, skip=None, limit=None):
+    """
+    searches Businesses
+    By passing in the appropriate options, you can search for available Businesses
+    :param searchString: pass an optional search string for looking up Businesses
+    :type searchString: str
+    :param skip: number of records to skip for pagination
+    :type skip: int
+    :param limit: maximum number of records to return
+    :type limit: int
+
+    :rtype: None
+    """
+    return "to be implemented"  # pragma: no cover  # pragma: no cover
+
+
+def businesses_post(party):
+    """
+    adds a reporting unit of type Business
+    Adds a new Business, or updates an existing Business based on the business reference provided
+    :param party: Business to add
+    :type party: dict | bytes
+
+    :rtype: None
+    """
+
+    v = validate.Validator(validate.Exists('id', 'reference'), validate.IsUuid('id'))
+    if not v.validate(party):
+        return make_response(jsonify(v.errors), 400)
+
+    party_uuid = party['id']
+    ru_ref = party['reference']
+    associations = party.get('associations')
+    address = party.get('address')
+
+    db_party = db.session.query(Party).filter(Party.party_uuid == party_uuid).first()
+
+    if db_party and not db_party.business:
+        return make_response(jsonify({'errors': "Existing party with '{}' does not identify a business."
+                                     .format(party_uuid)}), 400)
+
+    if not db_party:
+        db_party = Party(party_uuid)
+
+    # TODO: there's no attempt made to detect if an address already exists, just assumes new business means new address
+    business = db.session.query(Business).filter(Business.ru_ref == ru_ref).first()
+    if not business:
+        db_address = Address(**address)
+        business = Business(ru_ref, db_party, db_address)
+        db.session.add(business)
+
+    business.attributes = party.get('attributes', {})
+
+    if associations:
+        for assoc in associations:
+            assoc_type = assoc['sampleUnitType']
+            assoc_id = assoc['id']
+
+            assoc_party = db.session.query(Party).filter(Party.party_uuid == assoc_id).first()
+            # TODO: deal with missing party
+            # TODO: assumes a BI association, implied by the table BusinessRespondent, but can there be others?
+            business_respondent = BusinessRespondent()
+            business_respondent.respondent = assoc_party.respondent
+            business_respondent.business = business
+            # TODO: check the association doesn't already exist?
+            db.session.add(business_respondent)
+
+    db.session.commit()
```

Comment:
One for the future but is it worth thing about some kind of  session management. I implemented the context_manager one should on this page http://docs.sqlalchemy.org/en/latest/orm/session_basics.html